src/openzaak/components/documenten/tests/test_reserved_document.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) 2019 - 2020 Dimpact
+from datetime import date
+
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from openzaak.components.documenten.api.serializers import ReservedDocumentSerializer
+from openzaak.components.documenten.api.utils import generate_document_identificatie
+from openzaak.components.documenten.models import ReservedDocument
+from openzaak.components.documenten.tests.factories import (
+    EnkelvoudigInformatieObjectFactory,
+)
+
+
+class ReservedDocumentTests(APITestCase):
+
+    def setUp(self):
+        self.bronorganisatie = "812345678"
+        self.reserved_identificatie = "DOCUMENT-2025-0000000001"
+        self.new_identificatie = "DOCUMENT-2025-0000000002"
+        self.another_reserved_identificatie = "DOCUMENT-2025-0000000003"
+        self.url = reverse("reserveddocument-list", kwargs={"version": "1"})
+
+        self.reserved_document = ReservedDocument.objects.create(
+            identificatie=self.reserved_identificatie,
+            bronorganisatie=self.bronorganisatie,
+        )
+
+        ReservedDocument.objects.create(
+            identificatie=self.another_reserved_identificatie,
+            bronorganisatie=self.bronorganisatie,
+        )
+
+    def test_reserved_document_str_representation(self):
+        reserved_document = ReservedDocument.objects.create(
+            identificatie="DOCUMENT-2025-1234567890",
+            bronorganisatie="812345678",
+        )
+        expected_str = "812345678 - DOCUMENT-2025-1234567890"
+        self.assertEqual(str(reserved_document), expected_str)
+
+    def test_reserve_document_creates_successfully(self):
+        reserved_document = ReservedDocument.objects.create(
+            identificatie=self.new_identificatie, bronorganisatie=self.bronorganisatie
+        )
+        self.assertEqual(reserved_document.identificatie, self.new_identificatie)
+        self.assertEqual(reserved_document.bronorganisatie, self.bronorganisatie)
+
+    def test_generate_identificatie_excludes_reserved_and_issued_ids(self):
+        generated_id = generate_document_identificatie(
+            bronorganisatie=self.bronorganisatie,
+            date_value=date.today(),
+        )
+        self.assertNotIn(generated_id, [self.reserved_identificatie])
+
+    def test_create_enkelvoudig_info_object_with_reserved_identificatie(self):
+        eio = EnkelvoudigInformatieObjectFactory.create(
+            identificatie=self.reserved_identificatie,
+            bronorganisatie=self.bronorganisatie,
+        )
+
+        with self.assertRaises(ReservedDocument.DoesNotExist):
+            ReservedDocument.objects.get(
+                identificatie=self.reserved_identificatie,
+                bronorganisatie=self.bronorganisatie,
+            )
+
+        self.assertEqual(eio.identificatie, self.reserved_identificatie)
+
+    def test_create_enkelvoudig_info_object_with_non_reserved_identificatie(self):
```

Comment:
I don't think this test making the right assertions, it should only be possible to reserve an identificatie if it has not been reserved yet